server/storage/kv/etcd_kv.go

```diff
@@ -103,6 +109,55 @@ func (kv *etcdKVBase) Save(key, value string) error {
 	return nil
 }
 
+func (kv *etcdKVBase) GrantLease(ttlSeconds int64) (leaseID clientv3.LeaseID, err error) {
+	start := time.Now()
+	ctx, cancel := context.WithTimeout(kv.client.Ctx(), requestTimeout)
+	grantResp, err := kv.client.Grant(ctx, ttlSeconds)
+	cancel()
+	if err != nil {
+		e := errs.ErrEtcdGrantLease.Wrap(err).GenWithStackByCause()
+		log.Error("grant lease meet error",
+			zap.Int64("ttl", ttlSeconds),
+			errs.ZapError(e))
+		return 0, e
+	}
+	if cost := time.Since(start); cost > slowRequestTime {
+		log.Warn("lease grants too slow",
+			zap.Reflect("response", grantResp),
+			zap.Duration("cost", cost),
+			errs.ZapError(err))
+	}
+	log.Info("lease granted",
+		zap.Int64("lease-id", int64(grantResp.ID)),
+		zap.Int64("lease-timeout", ttlSeconds))
+	return grantResp.ID, nil
+}
+
+func (kv *etcdKVBase) SaveWithTTL(key, value string, ttlSeconds int64) error {
+	leaseID, err := kv.GrantLease(ttlSeconds)
```

Comment:
Could we use `etcdutil.EtcdKVPutWithTTL(context.TODO(),kv.client,key,value,ttlSeconds)` directly?

Reply:
Hmm, I was trying to put `SaveWithTTL` into a `SlowLogTxn` like other update methods for timeout and logging. But I guess that created more complexity.

Should we skip timeout / logging for this operation like other read methods?